plat/drivers/virtio/virtio_net.c

```diff
@@ -1139,6 +1139,9 @@ static int virtio_net_start(struct uk_netdev *n)
 	virtio_dev_drv_up(d->vdev);
 	uk_pr_info(DRIVER_NAME": %"__PRIu16" started\n", d->uid);
 
+	for (i = 0; i < d->rx_vqueue_cnt; i++)
+		virtqueue_host_notify(d->rxqs[i].vq);
+
```

Comment:
Hi @dinhngtu, We notify the device of receive queues when we call the `virtio_netdev_rx_fillup()`, which is called during `virtio_netdev_rx_queue_setup()`. Since this function is always called before `virtio_net_start()`, we don't need to notify the host again.

so you should remove this commit

Reply:
`virtio_netdev_rx_queue_setup` calls `virtio_netdev_rx_fillup` with `notify=0` which does not send the necessary notifications. These notifications must be sent later (after `DRIVER_OK`) which is why I've done it here.